### PR TITLE
USAGOV-1016-backup-cms-public - add CMS Public Files backup and restore capability

### PR DESCRIPTION
**Pull request by ghost**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-1016

## Checklist for the Developer

- [ ] A link to the JIRA ticket has been added above.
- [ ] The JIRA ticket identifies the desired result of this change.
- [ ] The JIRA ticket contains clear acceptance criteria.
- [ ] The JIRA ticket contains clear testing steps.
- [ ] The JIRA ticket contains a description of "Done".
- [ ] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [ ] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

### CODE REVIEW

File: bin/snapshot-backups/public-snapshot-create [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+# we might be running from a local dev machine
+SCRIPT_DIR="$(dirname "$0")"
+if [ -f $SCRIPT_DIR/env.local ]; then
+  . $SCRIPT_DIR/env.local
+fi
+if [ -f ./env.local ]; then
+  . ./env.local
+fi
+
+export BUCKET_SOURCE_FOLDER=public
```
Comment (akf) [inline bin/snapshot-backups/public-snapshot-create:18]: I probably gave you a bad steer on this one. While we have likely-looking files at public/, the files we want are actually in cms/public/. 

I tried uploading an image via the dev CMS and it created a new path under cms/public/ but not under public/. 

This probably has implications for the restore script as well.

**Review by akf — CHANGES_REQUESTED** [review]
I tested public-snapshot-list, public-snapshot-create, and public-snapshot-download in the dev space, using the stw helper. I have not yet tried doing an upload. 

I'm pretty sure the path for the public files on s3 is incorrect: it should be cms/public/. We have files in both places, but the ones under cms/public include newer files, so this must have been either a change, or the public/ path is just an artifact of someone's having made a mistake at some point.

File: bin/snapshot-backups/public-snapshot-create (hunk 1 above)
Comment (ghost) [reply bin/snapshot-backups/public-snapshot-create:18]: @akf - oh dang... that's not good.  I'll modify the paths.

**Comment by ghost** [comment]
This has been tested by doing backups/restores/downloads of the public files in Dev.

The incorrect path issue identified by @akf  has been fixed.

**Review by akf — APPROVED** [review]
I haven't tested all of this, but it does look safe to merge!

**Pull request merged**